internal/protcl/errors.go

```diff
@@ -101,3 +101,21 @@ func (e *ErrUnknownCommand) Error() string {
 func (ErrUnknownCommand) Recoverable() bool {
 	return true
 }
+
+// ErrProtocolType is for protocol type error
+type ErrProtocolType struct {
+	Type byte
+}
+
```

Comment:
If these errors should not disconnect client, you can implement Recoverable interface on them

Reply:
done